### ISSUE TITLE
Would be great to also have a raw JSON view

### ISSUE THREAD

**Issue by JeffCordes** [microsoft/BotFramework-Emulator#1694]
## Version
4.5.2

## Describe the bug
You can no longer view raw json from events and activities posted to and from the bot.  I understand the new tree view may be convenient and familliar to some, and would support it being a view option, but still keep the ability to view raw json.

## To Reproduce
1. Open bot emulator connect to a bot and send messages.
2. Try to copy the body of an event to try paste into another curl program like postman
3. notice there is no longer json, but a nested tree view

## Expected behavior
you can view raw json and copy/paste it

[bug]

**Comment by tonyanziano** [comment]
Hi @JeffCordes ,

Thanks for the input. Would something like a "Copy Raw JSON" button at the top of the JSON view be useful?

Or would you rather it had a tab / dropdown that allowed you to switch between raw and tree view?

**Comment by JeffCordes** [comment]
Thanks @tonyanziano 

I would prefer to have toggle to switch between views, as sometimes you would need to to copy only certain objects out of the json, like the entire from and recipient objects.

To give this some context, we are working on a bot that consumes many custom events (IEventActivity that has a custom data object in its value parameter),  The need for the json is really because there is no way to send a custom event in the emulator.  So the way to emulate the events is to copy an existing message json, manipulate it you be our payload, and post in in postman.

Perhaps in a future feature request the ability to send user defined events that automatically get all the correct user/conversation/etc info for the chat session could be useful.  like we supply a "name" and "value" (in json) and emulator can post that to the bot with all the conversation info.

we have rolled back versions and turned off updates for now.  unfortunately the new view is a blocker for us for now.

**Comment by tonyanziano** [comment]
@JeffCordes I can definitely see why that would be useful in your case.

We actually have had several asks for a feature that allows the user to send custom event payloads such as the one you described -- see #416 .

Thanks again for the feedback, I'll mention this to design.

**Comment by darrenj** [comment]
I just ran into this as well, a raw JSON view would be good but in the meantime a copy raw JSON button/option would go a long way. I use this a lot to retrieve a well formed Activity for other types of testing so the loss of the ability to copy/paste is an issue for this. 

For now you can workaround by saving transcript and opening in a text editor but not ideal :-)

**Comment by darrenj** [comment]
@JeffCordes Another idea is a middleware we wrote for the Virtual Assistant which you can find [here](https://github.com/microsoft/botframework-solutions/blob/master/lib/csharp/microsoft.bot.builder.solutions/microsoft.bot.builder.solutions/Middleware/EventDebuggerMiddleware.cs). Also covered briefly [here](https://github.com/microsoft/botframework-solutions/blob/master/docs/reference/assistant/events.md#event-debug-middleware)

This enables you to send events by sending a message like this through the Emulator:

```
/event:{ "Name": "VA.Location", "Value": "47.639620,-122.130610" }
```

**Comment by JeffCordes** [comment]
Thanks for recommendation @darrenj.  

we do have a similar "slash command" middleware that looks for messages commands that start with / and executes some things for testing.  That middleware is only registered in startup.cs if the if(!env.IsProduction()) for security reasons.  This covers a lot of our scenarios.

We have a few edge cases with custom ChanelData parameters on a custom event that tie into our user/security/environment workflow.  Though i guess we could expand the object sent in to catch both.  

It is also currently not easy to send custom channel data parameters in the emulator.

**Comment by cwhitten** [comment]
Hi! We'll add something in our next sprint (4.6) to make it easy to copy the JSON data.

**Comment by tonyanziano** [comment]
Addressed in #1801. Look for this feature in our [latest nightly build](https://github.com/microsoft/BotFramework-Emulator-nightlies/releases) tonight after around 10:30 PM PST.
